### PR TITLE
DOCS-878: Apply Matt Vella's feedback to extra params

### PR DESCRIPTION
**Pull request by sguequierre**
* Switch "utilize" and "define" use cases
* Change wording to make this more clear that this is meant to be used by users who implement modular components (later on should go with broader changes when registry is ready) 
* Hide "utilize" code steps in expand

### CODE REVIEW

File: docs/program/use-extra-params.md [hunk 1]
```diff
@@ -10,76 +10,18 @@ aliases:
...
+You can use `extra` parameters with modular {{< glossary_tooltip term_id="resource" text="resource" >}} implementations that are new *models* of built-in resource types.
 
-To do this, you must code your own modified implementation of the resource type's API for a model.
-See [Extend Viam with Modular Resources](/extend/modular-resources/) for more information and [instructions](/extend/modular-resources/#use-a-modular-resource-with-your-robot) on modifying API specifications.
+For example, a new model of [sensor](/components/sensor/), or a new model of [SLAM Service](/services/slam/).
 
-An example of how to check the values of keys in an `extra` parameter of a [built-in resource API method](/program/apis/), the sensor component's [Readings](/components/sensor/#readings):
+The `extra` parameters in that built-in resource type's [API](/extend/modular-resources/#apis) give users the option to pass information to a resource's driver that isn't specified as a parameter for all models of the resource type.
+This is necessary to keep the API of resource types consistent across, for example, all models of [motor](/components/motor/) or all models of [camera](/components/camera).
```
Comment (sguequierre) [inline docs/program/use-extra-params.md:22]:
```suggestion
This is necessary to keep the API of resource types consistent across, for example, all models of [motor](/components/motor/) or all models of [camera](/components/camera/).
```

File: docs/program/use-extra-params.md [hunk 2]
```diff
@@ -10,76 +10,18 @@ aliases:
   - "/program/sdks/use-extra-params"
 ---
 
-How to [utilize](#utilize) and [define](#define) the `extra` parameters that many {{< glossary_tooltip term_id="resource" text="resource" >}} API methods offer in the Go and Python SDKs.
+How to [utilize](#utilize) and [define](#define) the `extra` parameters that many {{< glossary_tooltip term_id="resource" text="resource" >}} [API methods](/program/apis/) offer in the Go and Python SDKs.
 
 ## Utilize
 
-Use `extra` parameters to pass information to a {{< glossary_tooltip term_id="resource" text="resource's" >}} driver that is not explicitly specified as a parameter in the [built-in resource type's API specification](/extend/modular-resources/#apis).
+You can use `extra` parameters with modular {{< glossary_tooltip term_id="resource" text="resource" >}} implementations that are new *models* of built-in resource types.
```
Comment (mcvella) [inline docs/program/use-extra-params.md:17]:
```suggestion
You can use `extra` parameters with modular {{< glossary_tooltip term_id="resource" text="resource" >}} implementations that are *models* of built-in resource types.
```
Don't think we need the word 'new', as I guess any modular resource is 'new'?

File: docs/program/use-extra-params.md [hunk 3]
```diff
@@ -10,76 +10,18 @@ aliases:
...
 
-To do this, you must code your own modified implementation of the resource type's API for a model.
-See [Extend Viam with Modular Resources](/extend/modular-resources/) for more information and [instructions](/extend/modular-resources/#use-a-modular-resource-with-your-robot) on modifying API specifications.
+For example, a new model of [sensor](/components/sensor/), or a new model of [SLAM Service](/services/slam/).
 
-An example of how to check the values of keys in an `extra` parameter of a [built-in resource API method](/program/apis/), the sensor component's [Readings](/components/sensor/#readings):
+The `extra` parameters in that built-in resource type's [API](/extend/modular-resources/#apis) give users the option to pass information to a resource's driver that isn't specified as a parameter for all models of the resource type.
```
Comment (mcvella) [inline docs/program/use-extra-params.md:21]:
```suggestion
The `extra` parameters in that built-in resource type's [API](/extend/modular-resources/#apis) allow users to pass information to a resource's driver that isn't specified as a parameter for all models of the resource type.
```

**Review by mcvella — APPROVED** [review]
lgtm with two small potential changes

**Pull request merged**